LambdaS3FileZipper/FileZipper.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
+using LambdaS3FileZipper.Interfaces;
+
+namespace LambdaS3FileZipper
+{
+	public class FileZipper : IFileZipper
+	{
+		public async Task<string> Compress(string localDirectory)
+		{
+			if (!Directory.Exists(localDirectory))
+			{
+				throw new IOException($"{localDirectory} does not exist");
+			}
+
+			var directoryName = Path.GetDirectoryName(localDirectory);
+			var zipPath = Path.Combine(localDirectory, $"{directoryName}.zip");
```

Comment:
Shouldn't we invert `localDirectory` and `directoryName` here, given that `directoryName` is the parent directory?